docs/advanced.md

```diff
@@ -221,6 +221,28 @@ with httpx.Client(headers=headers) as client:
     ...
 ```
 
+## Monitoring download progress
+
+If you need to monitor download progress of large responses, you can use stream and the `response.last_raw_chunk_size` property.
+
+For example, showing a progress bar using the [`tqdm`](https://github.com/tqdm/tqdm) library while a response is being downloaded could be done like this…
+
+```python
+import tempfile
+
+import httpx
+from tqdm import tqdm
+
+with tempfile.NamedTemporaryFile() as download_file:
+    data = b"@" * 1000000
+    with httpx.stream("POST", "https://httpbin.org/anything", data=data) as response:
+        content_length = int(response.headers["Content-Length"])
```

Comment:
I *guess* we ought to also cover the case where `Content-Length` isn't present.

Perhaps something like:

```python
if "Content-Length" in response.headers:
    total = int(response.headers["Content-Length"])
else:
    total = None
```

Reply:
I didn't think about that, but you are right